light_node/src/main.rs

```diff
@@ -518,85 +568,81 @@ fn main() {
     ];
 
-    let rocks_db = match open_kv(&env.storage.db_path, schemas, &env.storage.db_cfg) {
-        Ok(db) => Arc::new(db),
+    let storages = match initialize_databases(&cache, &env) {
         Err(e) => shutdown_and_exit!(
-            error!(log, "Failed to create RocksDB database at '{:?}'", &env.storage.db_path; "reason" => e),
+            error!(
+                log,
+                "Failed to create initialize RocksDB databases '{:?}'", e
+            ),
             actor_system
         ),
+        Ok(db) => db,
     };
-    debug!(log, "Loaded RocksDB database");
 
-    match check_database_compatibility(rocks_db.clone(), DATABASE_VERSION, &tezos_env, &log) {
-        Ok(false) => shutdown_and_exit!(
-            crit!(log, "Database incompatibility detected"),
-            actor_system
-        ),
-        Err(e) => shutdown_and_exit!(
-            error!(log, "Failed to verify database compatibility"; "reason" => e),
-            actor_system
-        ),
-        _ => (),
+    for db in storages.iter() {
+        match check_database_compatibility(db.clone(), DATABASE_VERSION, &tezos_env, &log) {
+            Ok(false) => shutdown_and_exit!(
+                crit!(log, "Database incompatibility detected"),
+                actor_system
+            ),
+            Err(e) => shutdown_and_exit!(
+                error!(log, "Failed to verify database compatibility"; "reason" => e),
+                actor_system
+            ),
+            _ => (),
+        };
     }
 
-    let schemas = vec![BlockStorage::descriptor()];
-
     {
-        let commit_logs = match open_cl(&env.storage.db_path, schemas) {
+        let commit_logs = match open_cl(&env.storage.db_path, vec![BlockStorage::descriptor()]) {
             Ok(commit_logs) => Arc::new(commit_logs),
             Err(e) => shutdown_and_exit!(
                 error!(log, "Failed to open commit logs"; "reason" => e),
                 actor_system
             ),
         };
-
-        let persistent_storage = PersistentStorage::new(rocks_db, commit_logs);
-        let tezedge_context = TezedgeContext::new(
-            BlockStorage::new(&persistent_storage),
-            persistent_storage.merkle(),
-        );
-        match resolve_storage_init_chain_data(
-            &tezos_env,
-            &env.storage.db_path,
-            &env.storage.tezos_data_dir,
-            &env.storage.patch_context,
-            &log,
-        ) {
-            Ok(init_data) => block_on_actors(
-                env,
-                tezos_env,
-                init_data,
-                Arc::new(tezos_identity),
-                actor_system,
-                persistent_storage,
-                tezedge_context,
-                log,
-            ),
-            Err(e) => shutdown_and_exit!(
-                error!(log, "Failed to resolve init storage chain data."; "reason" => e),
-                actor_system
-            ),
+        if let [kv, kv_context, kv_actions] = &storages[..] {
```

Comment:
@mat13mn 
what if we remove `&` here,
than we dont need to copy kv.clone(),
or something like 'to give owner ship to PersistentStorage"

Reply:
i tried - results with compilation error
```
error[E0508]: cannot move out of type `[Arc<DB>]`, a non-copy slice
   --> light_node/src/main.rs:610:47
    |
610 |         if let [kv, kv_context, kv_actions] = storages[..] {
    |                 --  ----------  ----------    ^^^^^^^^^^^^
    |                 |   |           |             |
    |                 |   |           |             cannot move out of here
    |                 |   |           |             help: consider borrowing here: `&storages[..]`
    |                 |   |           ...and here
    |                 |   ...and here
    |                 data moved here
    |
    = note: move occurs because these variables have types that don't implement the `Copy` trait

```

Reply:
ok, no problem